frontend/app-development/layout/version-control/VersionControlHeader.tsx

```diff
@@ -157,22 +160,26 @@ export const VersionControlHeader = (props: IVersionControlHeaderProps) => {
       });
     }
   };
-  const repoPushMutation = useRepoPushMutation(org, app);
+
+  const repoPushMutation = useRepoPushMutation(org, app, setShouldShowSpinner);
   const pushChanges = async () => {
+    setShouldShowSpinner(true);
     setModalState({
       ...initialModalState,
       header: t('sync_header.sharing_changes'),
       isLoading: true,
     });
-    await repoPushMutation.mutate();
-    setHasChangesInMaster(false);
-    setHasChangesInLocalRepo(false);
+    await repoPushMutation.mutateAsync();
+
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
So this line (the timeout) and the one below should be removed. It is a hack that allows things to appear work as expected, but it shouldn't be needed once the `onSuccess`-method sets `setShouldShowSpinner(false)`.

Reply:
removed